test/hotel_organizer_test.rb

```diff
@@ -0,0 +1,102 @@
+require_relative 'test_helper'
+
+describe "HotelOrganizer Class" do  
+  before do
+    @hotel_organizer = Hotel::HotelOrganizer.new
+    
+    @reservation1 = @hotel_organizer.make_reservation(18, Date.new(2018,01,01), Date.new(2018,01,06), nil)
+    @reservation2 = @hotel_organizer.make_reservation(13, Date.new(2018,01,03), Date.new(2018,01,12), nil)
+    @reservation3 = @hotel_organizer.make_reservation(15, Date.new(2018,02,02), Date.new(2018,02,10), nil)
+    
+    @hotel_organizer.add_reservation(@reservation1)
+    @hotel_organizer.add_reservation(@reservation2)
+    @hotel_organizer.add_reservation(@reservation3)
+  end 
+  
+  describe "initialize" do
+    it "is an instance of HotelOrganizer" do
+      expect(@hotel_organizer).must_be_kind_of Hotel::HotelOrganizer    
+    end 
+    
+    it "checks data structures at initialize" do
+      expect(@hotel_organizer.rooms).must_be_kind_of Array
+      expect(@hotel_organizer.reservations).must_be_kind_of Array
+    end
+  end 
+  
+  describe "method total_rooms" do 
+    it "method returns an array" do
+      rooms = @hotel_organizer.total_rooms
+      expect(rooms).must_be_kind_of Array
+    end
+    
+    it "rooms array contains 20 rooms" do
+      expect(@hotel_organizer.rooms.length).must_equal 20
+    end
+  end
+  
+  describe "method make_reservation" do
+    it "returns an instance of Reservation" do
+      expect(@reservation1).must_be_kind_of Hotel::Reservation
+    end
+  end 
+  
+  describe "method add_reservation" do   
```

Comment:
It would be good to add tests here that test for different types of reservation conflicts and edge cases for allowed overlapping reservations.

Reply:
Fixed bug to make room show as available also on the last_day of reservation, made new test that's passing.